crates/analytics/src/payments/types.rs

```diff
@@ -104,6 +104,11 @@ where
                 .add_filter_in_range_clause(PaymentDimensions::ErrorReason, &self.error_reason)
                 .attach_printable("Error adding error reason filter")?;
         }
+        if !self.first_attempt.is_empty() {
+            builder
+                .add_filter_in_range_clause("first_attempt", &self.first_attempt)
```

Comment:
Why aren't we using a `PaymentDimensions` enum variant as the first param here?

Reply:
This is because, first_attempt is added only as a filter in `PaymentFilters` and not in `PaymentDimensions`. Basically, this field can only be added in the filters and not as a `group_by`.

Reply:
We are also planning to segregate these structs based on the type of sources we have , but for now this is needed.